tests/queries/0_stateless/01763_max_distributed_depth.sql

```diff
@@ -0,0 +1,18 @@
+DROP TABLE IF EXISTS tt6;
+
+CREATE TABLE tt6
+(
+	`id` UInt32,
+	`first_column` UInt32,
+	`second_column` UInt32,
+	`third_column` UInt32,
+	`status` String
+
+)
+ENGINE = Distributed('test_shard_localhost', '', 'tt6', rand());
+
+INSERT INTO tt6 VALUES (1, 1, 1, 1, 'ok'); -- { serverError 580 }
```

Comment:
We probably also need a test with the max_depth disabled, so that it returns stack overflow as before.